lib/spacesuit/proxy_handler.ex

```diff
@@ -117,26 +116,26 @@ defmodule Spacesuit.ProxyHandler do
     {ip, _port} = peer
 
     ip
-      |> Tuple.to_list
-      |> Enum.map(&(Integer.to_string(&1)))
-      |> Enum.join(".")
+    |> Tuple.to_list()
+    |> Enum.map(&Integer.to_string(&1))
+    |> Enum.join(".")
   end
 
   # Take static headers from the config and add them to this request
   def add_headers_to(headers, added_headers) do
-    Map.merge((headers || %{}), (added_headers || %{}))
+    Map.merge(headers || %{}, added_headers || %{})
   end
 
   # Convery headers from Cowboy map format to Hackney list format
   def cowboy_to_hackney(headers, peer, url) do
-    [[_, _, host | _ ] | _ ] = url
+    [[_, _, host | _] | _] = url
 
     (headers || %{})
-      |> Map.put("X-Forwarded-For", peer)
-      |> Map.put("X-Forwarded-Url", url)
-      |> Map.put("X-Forwarded-Host", host)
-      |> Map.drop([ "host", "Host" ])
-      |> Map.to_list
+    |> Map.merge(%{"x-forwarded-for" => peer}, fn _k, a, b -> "#{a}, #{b}" end)
```

Comment:
Does this necessarily maintain the order of the ips?

eg:
```

Would

"X-Forwarded-For: <client>, <proxy1>, <proxy2>" 
+ 
"x-forwarded-For: <other>"
= 
"x-forwarded-for = <client>, <proxy1>, <proxy2>, <other>"

OR

x-forwarded-for = <other>, <client>, <proxy1>, <proxy2>
```
I guess the question is, what governs the ordering here?

Reply:
Apologies for formatting!

Reply:
I think it will do the right thing, it seems to maintain order. Thanks!